tests/unit/test_search_presenters.py

```diff
@@ -109,37 +112,800 @@ class TestSearchSummary(unittest.TestCase):
+            'plural': 'datacentre tiers'
+        }
+        summary_fragment = SummaryFragment(
+            id, filters, self.rules_instance_mock)
+        self.assertEqual(
+            summary_fragment.str(),
+            u"datacentre tiers <em>TIA-942 Tier 1</em>" +
+            u" or <em>uptime institute tier 1</em>")
+
+    def test_fragment_with_label_and_three_filters(self):
+        id = 'Datacentre tier'
+        filters = [
+            'TIA-942 Tier 1',
+            'uptime institute tier 1',
+            'uptime institute tier 2']
+        self.rules['label'] = {
+            'singular': 'datacentre tier',
+            'plural': 'datacentre tiers'
+        }
+        summary_fragment = SummaryFragment(
+            id, filters, self.rules_instance_mock)
+        self.assertEqual(
+            summary_fragment.str(),
+            u"datacentre tiers <em>TIA-942 Tier 1</em>" +
+            u", <em>uptime institute tier 1</em>" +
+            u" or <em>uptime institute tier 2</em>")
+
+    def test_fragment_with_label_and_three_filters_with_prepositions(self):
+
+        def _add_preposition(filter_id=None, filter_string=None):
+            if filter_id == 'TIA-942 Tier 1':
+                preposition = 'with a'
+            elif filter_id == 'uptime institute tier 1':
+                preposition = 'met with a'
+            elif filter_id == 'uptime institute tier 2':
+                preposition = 'aligned with'
+            return u"{} {}".format(preposition, filter_string)
+
+        id = 'Datacentre tier'
+        filters = [
+            'TIA-942 Tier 1',
+            'uptime institute tier 1',
+            'uptime institute tier 2']
+        self.rules['label'] = {
+            'singular': 'datacentre tier',
+            'plural': 'datacentre tiers'
+        }
+        self.rules['filterRules'] = []  # filterRules just needs to be set
+        self.rules_instance_mock.filter_rules_ids = filters
+        self.rules_instance_mock.add_filter_preposition = _add_preposition
+        summary_fragment = SummaryFragment(
+            id, filters, self.rules_instance_mock)
+        self.assertEqual(
+            summary_fragment.str(),
+            u"datacentre tiers with a <em>TIA-942 Tier 1</em>" +
+            u", met with a <em>uptime institute tier 1</em>" +
+            u" or aligned with <em>uptime institute tier 2</em>")
+
+    def test_str_method_works(self):
+        id = 'Datacentre tier'
+        filters = ['TIA-942 Tier 1']
+        summary_fragment = SummaryFragment(
+            id, filters, self.rules_instance_mock)
+        summary_fragment.label_string = u"categories"
+        summary_fragment.filters = [u"FreeOption"]
+        self.rules['filtersPreposition'] = 'with'
+        self.assertEqual(summary_fragment.str(), u"categories with FreeOption")
+
+
+filter_groups = [
+    {
+        'depends_on_lots': ['scs', 'saas', 'iaas'],
+        'filters': [{
+            'id': 'serviceTypes-implementation',
```

Comment:
Is all this stuff a duplicate from the YAML files? Can we import that instead? Save duplication, breaks if changes, and so on.